cpu/efm32/periph/spi.c

```diff
@@ -56,6 +59,66 @@ void spi_init_pins(spi_t bus)
     gpio_init(spi_config[bus].miso_pin, GPIO_IN_PD);
 }
 
+spi_clk_t spi_get_clk(spi_t bus, uint32_t freq)
+{
+    (void)bus;
+
+    /* bound divider to 65536 */
+    if (freq < DIV_ROUND_UP(CLOCK_CORECLOCK, 65536)) {
+        return (spi_clk_t){ .err = -EDOM };
+    }
+
+    /* =================================================================
+     * from USART_BaudrateSyncSet() of the gecko SDK
+     */
+    uint32_t ref_freq, clkdiv;
+
+    /* Prevent dividing by 0. */
+    EFM_ASSERT(freq);
+
+    /*
+     * CLKDIV in synchronous mode is given by:
+     *
+     * CLKDIV = 256 * (fHFPERCLK/(2 * br) - 1)
+     */
+
+    /* HFPERCLK/HFPERBCLK used to clock all USART/UART peripheral modules. */
+#if defined(_SILICON_LABS_32B_SERIES_2)
+    ref_freq = CMU_ClockFreqGet(cmuClock_PCLK);
+#else
+#if defined(_CMU_HFPERPRESCB_MASK)
+    if (dev == USART2) {
+        ref_freq = CMU_ClockFreqGet(cmuClock_HFPERB);
+    }
+    else {
+        ref_freq = CMU_ClockFreqGet(cmuClock_HFPER);
+    }
+#else
+    ref_freq = CMU_ClockFreqGet(cmuClock_HFPER);
+#endif
+#endif
+
+    clkdiv = (ref_freq - 1) / (2 * freq);
+    clkdiv = clkdiv << 8;
+
+    /* =================================================================
+     *
+     * The manual say that `the clock division factor have a 15-bit
+     * integral part and a 5-bit fractional part` but only the integral
+     * part was used in the gecko SDK.
+     */
+
+    /* br = fHFPERCLK/(2 x (1 + USARTn_CLKDIV / 256)) */
+    return (spi_clk_t){ .clk = DIV_ROUND_UP(ref_freq, (2 * (1 + clkdiv / 256))) };
+}
+
+int32_t spi_get_freq(spi_t bus, spi_clk_t clk)
+{
+    (void)bus;
+    if (clk.err) { return -EINVAL; }
```

Comment:
```suggestion
    if (clk.err) {
        return -EINVAL;
    }
```
Here and at dozends other places.